staging/src/k8s.io/api/auditregistration/v1alpha1/types.go

```diff
@@ -192,3 +219,107 @@ type ServiceReference struct {
 	// +optional
 	Path *string `json:"path,omitempty" protobuf:"bytes,3,opt,name=path"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AuditClass is a set of rules that categorize requests
+//
+// This should be considered a highly privileged object, as modifying it
+// will change what is logged.
+type AuditClass struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec is the spec for the audit class
+	Spec AuditClassSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AuditClassList is a list of AuditClass items
+type AuditClassList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// List of audit classes
+	Items []AuditClass `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// AuditClassSpec is the spec for the audit class
+type AuditClassSpec struct {
+	// RequestSelectors defines a list of RequestSelectors
+	RequestSelectors []RequestSelector `json:"requestSelectors,omitempty" protobuf:"bytes,1,opt,name=requestSelectors"`
+}
+
+// RequestSelector selects requests by matching on the given fields. Selectors are
+// used to compose audit classes.
+type RequestSelector struct {
+	// The users (by authenticated user name) in this attribute group.
+	// An empty list implies every user.
```

Comment:
Document: how many users max should be in this list? If you have more than that, what should you do?